### DISCUSSION TITLE
Record dB for only specified Frequenc.

### DISCUSSION THREAD

**Discussion by SnipeyBB** [stas-sl/esphome-sound-level-meter#30] [Q&A]
Hello, it is possible to record only the dB for 100 Hz or for a specifig Range like 98 - 102 Hz?

Thank you

**Comment by stas-sl** [comment]
Hi, should be possible, it all comes down to designing corresponding digital filter, but you need some DSP skills/knowledge to be able to do that correctly. I wouldn't call myself expert in this topic, so I can't quickly provide you with ready coefficients that you can just copy-paste into config, but you'll have to spend some time familiarising with this topic. 

There are some online tools like https://arachnoid.com/BiQuadDesigner/index.html, in which you can design filters like that

<img width="946" alt="image" src="https://github.com/user-attachments/assets/86b3d68e-d7bf-4c70-a571-db3cf4900331">

Though, I haven't used coefficients from this tool myself, so I can't guarantee they will work as is. Or you can use any other tool like MATLAB/Jupyter Notebook+scipy to design this filter. 

If you are familiar with Python, you can also checkout [my notebook](https://github.com/stas-sl/esphome-sound-level-meter/blob/main/math/filter-design.ipynb), where I explain how to make sure that filter produces the desired result.
